feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
```

Comment:
I tried to quickly look into this class, and this is something I came up with. Please bear in mind that:
- is the waitingTime from `this.config?.updateConfigurationMessage?.waitingTime` a requirement that has to be met? If that's the case, this solution of mine is not complete.
- I might've missunderstood the idea behind this code. If so, let's discuss it
- I haven't tested it, and it might not work

That being said, here's the TS file:
```ts
import { Component } from '@angular/core';
import { ConfiguratorCommonsService } from '@spartacus/core';
import { ConfigRouterExtractorService } from '@spartacus/storefront';
import { Observable } from 'rxjs';
import { map, switchMap } from 'rxjs/operators';

@Component({
  selector: 'cx-config-message',
  templateUrl: './config-message.component.html',
})
export class ConfigMessageComponent {
  protected hasPendingChanges$: Observable<
    Boolean
  > = this.configRouterExtractorService
    .extractRouterData()
    .pipe(
      switchMap((routerData) =>
        this.configuratorCommonsService
          .hasPendingChanges(routerData.owner)
          .pipe(
            switchMap((hasPendingChanges) =>
              this.configuratorCommonsService
                .isConfigurationLoading(routerData.owner)
                .pipe(map((isLoading) => hasPendingChanges || isLoading))
            )
          )
      )
    );

  visibleClass$: Observable<string> = this.hasPendingChanges$.pipe(
    map((hasPendingChangesOrIsLoading) =>
      hasPendingChangesOrIsLoading ? 'd-none' : ''
    )
  );

  constructor(
    protected configuratorCommonsService: ConfiguratorCommonsService,
    protected configRouterExtractorService: ConfigRouterExtractorService
  ) {}
}

```

and here's the html template:
```html
<div class="wrapper">
  <diV id="cx-config-update-message" [class]="visibleClass$ | async">
    <cx-spinner></cx-spinner>
    <strong>{{ 'configurator.header.updateMessage' | cxTranslate }}</strong>
  </diV>
</div>

```

Reply:
I think we need to discuss the whole component with regards to setting a timeout and the DOM manipulation.
As I only moved the component and I was not the original developer I'd like to involve my colleague Lukas (@schuppell) here.

Let's have a call next week regarding this.
As an upfront information: this component shows some kind of loading message ("Configuration is updated wait a little bit!") if the calculation of the configuration-state in the configuration-engine in the backend takes a little bit longer (e.g. due to complexity of calculation or low bandwidth).
In order to avoid that this message is constantly flickering although the response time of the engine is good, we decided to show the message only after a certain time if the calculation is still not finished. Thus the timeout.

Reply:
Ok, sounds good. If you want to add some delay before showing the message, there are rxjs operators that we can consider for this case.

Reply:
We have changed the config-message.component according to your latest suggestions. Please review.